src/config.ts

```diff
@@ -53,6 +53,19 @@ export type ProviderConfig = {
     /* An optional function to generate the bucket asset key. */
     generateAssetKey?: (filePathOrURL: string, folder: string) => string;
   };
+
+  'cloudflare-r2'?: {
+    endpoint: string;
+    accountId: string;
+    bucket?: string;
+    bucketUrlPublic?: string;
```

Comment:
is there any way `bucketUrlPublic` can be constructed from endpoint, accountId and bucket?

Reply:
Without a Cloudflare API Token (R2_CF_API_TOKEN) to my knowledge no.

There are two ways for public access:
- r2.dev subdomain
- custom domain
Both must be enabled first on a bucket level.

If a Cloudflare API Token is given to the provider the r2.dev subdomain will be enabled when creating the bucket (if none is given). Making it usable instantly without manual interaction.